packages/ag-charts-enterprise/src/features/sync/chartSync.ts

```diff
@@ -125,39 +125,48 @@ export class ChartSync extends BaseProperties implements _ModuleSupport.ModuleIn
-        for (const [chart, axis] of syncManager.getGroupSiblingAxes(this.groupId)) {
+        for (const chart of syncManager.getGroupSiblings(this.groupId)) {
             if (!chart.modulesManager.getModule<ChartSync>('sync')?.nodeInteraction) continue;
-            if (!CartesianAxis.is(axis) || axis.direction !== mainDirection) continue;
-
-            // Find matching nodes for the main direction.
-            let matchingNodes = chart.series
-                .filter((s) => s.visible)
-                .map(this.findMatchingNodes(axis, mainDirection, valueIsDate, eventValue))
-                .filter(isDefined);
-
-            // Narrow matches by matching the secondary direction key.
-            if (useSecondaryDirectionKey) {
-                const secondaryKey = `${secondaryDirection}Key` as const;
-                const secondaryValue = _ModuleSupport.isObjectWithProperty(event.currentHighlight, secondaryKey)
-                    ? event.currentHighlight?.[secondaryKey]
-                    : undefined;
-
-                matchingNodes = matchingNodes.filter(({ nodeDatum }) => {
-                    return (
-                        isObjectWithStringProperty(nodeDatum, secondaryKey) &&
-                        nodeDatum[secondaryKey] === secondaryValue
-                    );
-                });
+
+            let dispatched = false;
+            for (const axis of chart.axes) {
+                if (!CartesianAxis.is(axis) || axis.direction !== mainDirection) continue;
+
+                // Find matching nodes for the main direction.
+                let matchingNodes = chart.series
+                    .filter((s) => s.visible)
+                    .map(this.findMatchingNodes(axis, mainDirection, valueIsDate, eventValue))
+                    .filter(isDefined);
+
+                // Narrow matches by matching the secondary direction key.
+                if (useSecondaryDirectionKey) {
+                    const secondaryKey = `${secondaryDirection}Key` as const;
+                    const secondaryValue = _ModuleSupport.isObjectWithProperty(event.currentHighlight, secondaryKey)
+                        ? event.currentHighlight?.[secondaryKey]
+                        : undefined;
+
+                    matchingNodes = matchingNodes.filter(({ nodeDatum }) => {
+                        return (
+                            isObjectWithStringProperty(nodeDatum, secondaryKey) &&
+                            nodeDatum[secondaryKey] === secondaryValue
+                        );
+                    });
+                }
+
+                if (
+                    matchingNodes.length === 1 &&
+                    matchingNodes[0]?.nodeDatum !== chart.ctx.highlightManager.getActiveHighlight()
+                ) {
+                    const { nodeDatum } = matchingNodes[0] ?? {};
+                    this.dispatchHighlightUpdate(chart, nodeDatum);
```

Comment:
super nitty nit, I wouldn't destruct here, it's a waste of an object 😅

Reply:
Done, good spot